src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -542,14 +542,13 @@ public String Insert(int startIndex, String value)
             return result;
         }
 
-        public static string Join(char separator, params string[] value)
+        public unsafe static string Join(char separator, params string[] value)
         {
             if (value == null)
             {
                 throw new ArgumentNullException(nameof(value));
             }
-
-            return Join(separator, value, 0, value.Length);
+            return JoinCore(&separator, 1, value, 0, value.Length);
```

Comment:
If we add some checks or something else inside `Join(separator, value, 0, value.Length)` later, then JIT compiler should make call not inline. So I suggests use `JoinCore(&separator, 1, value, 0, value.Length)` here. Also code looks more monotonous in Join methods.